source/extensions/filters/common/rbac/matchers.h

```diff
@@ -163,13 +163,17 @@ class PortMatcher : public Matcher {
 class AuthenticatedMatcher : public Matcher {
 public:
   AuthenticatedMatcher(const envoy::config::rbac::v2alpha::Principal_Authenticated& auth)
-      : name_(auth.name()) {}
+      : name_(auth.name()),
+        matcher_(auth.has_principal_name()
+                     ? absl::make_optional<Matchers::StringMatcher>(auth.principal_name())
```

Comment:
Nit: probably could just use `{}` initialization here, but this is fine.

Reply:
I tried but it failed with error:
```
error: initializer list cannot be used on the right hand side of operator '?'
                     ? {auth.principal_name()}
                     ^ ~~~~~~~~~~~~~~~~~~~~~~~
```